src/Symfony/Contracts/HttpClient/ApiClientInterface.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\HttpClient;
+
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\Test\ApiClientTestCase;
+
+/**
+ * Provides flexible methods for interacting with HTTP APIs.
+ *
+ * Unless otherwise specified, implementations MUST send a JSON-compatible
+ * "accept" header - typically "application/json".
```

Comment:
Unless specified otherwise in the 'headers' $option, implementations MUST ...

Reply:
actually, why? if the client must be for json, why is there an interface? are there more meaningful implementations than json_decode?

what if i want to implement this for an xml api? (not soap, there array would make little sense as a response type, but some generic xml thing)

Reply:
Except `toArray()`, which doesn't fit other content types, the interface fits xml yes.
The idea here is to tell ppl that if they don't explicitly an `Accept` header, then they can expect application/json to be sent for them. Then, if one client is configured to send xml, one should ensure that such a client would be injected only where that default is expected.
Would that make sense? How would you formulate that?

Reply:
Reformulated.Bette?

Reply:
yes!